Libplanet/Net/Swarm.cs

```diff
@@ -711,6 +707,179 @@ public void BroadcastTxs(IEnumerable<Transaction<T>> txs)
             }
         }
 
+        internal string TraceTable()
+        {
+            if (_protocol is null)
+            {
+                return string.Empty;
+            }
+            else
+            {
+                return _protocol.Trace();
+            }
+        }
+
+        internal async Task AddPeersAsync(
+            IEnumerable<Peer> peers,
+            TimeSpan? timeout,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (_protocol is null)
+            {
+                throw new ArgumentNullException(nameof(_protocol));
+            }
+
+            if (cancellationToken == default(CancellationToken))
+            {
+                cancellationToken = _cancellationToken;
+            }
+
+            try
+            {
+                KademliaProtocol<T> kp = (KademliaProtocol<T>)_protocol;
+
+                List<Task> tasks = new List<Task>();
+                foreach (Peer peer in peers)
+                {
+                    tasks.Add(kp.PingAsync(
+                        peer,
+                        timeout: timeout,
+                        cancellationToken: cancellationToken));
+                }
+
+                await Task.WhenAll(tasks);
+            }
+            catch (TimeoutException)
```

Comment:
I think that it's useless because `Take.WhenAll` wraps its exceptions into `AggregateException`.

Reply:
Actually, `AggregateException` will be unwrapped in this case.
Please take a look at https://stackoverflow.com/questions/12007781/why-doesnt-await-on-task-whenall-throw-an-aggregateexception

Reply:
I'd confused with `.Wait()`.